### PR TITLE
Add more tests and jsdocs for saltcorn-admin-models & Add ability to limit file types to upload

### PR DESCRIPTION
**Pull request by AleksandrSokolov**
1. Add more tests and jsdocs for saltcorn-admin-models
2. Add ability to limit file types to upload

### CODE REVIEW

**Comment by glutamate** [comment]
@AleksandrSokolov it would be better to add the file filter to the upload fileview here:  

https://github.com/saltcorn/saltcorn/blob/v0.8.0-beta.0/packages/saltcorn-data/base-plugin/fileviews.js#L101-L128

Add your field (https://github.com/saltcorn/saltcorn/pull/1472/files#diff-b67356cdcfedae3f65d51bff81d7350911b0be04245d24021b5b9feac7c983afR597-R602) here instead: https://github.com/saltcorn/saltcorn/blob/v0.8.0-beta.0/packages/saltcorn-data/base-plugin/fileviews.js#L113

and your implementation of the accepts attr to the input here: https://github.com/saltcorn/saltcorn/blob/v0.8.0-beta.0/packages/saltcorn-data/base-plugin/fileviews.js#L124

The admin should be able to upload any kind of file, but restricting the user is very useful (e.g. only images)

**Comment by AleksandrSokolov** [comment]
@glutamate please review. I reworked file filtering. Added accept filter to file field. So for every field column possible to define own filter. Also default filter applied if user does not define filter for particular field.

File: packages/server/routes/fields.js [hunk 1]
```diff
@@ -310,6 +312,15 @@ const fieldFlow = (req) =>
                     "Deleting a row will also delete the file referenced by this field"
                   ),
                 },
+                {
```
Comment (glutamate) [inline packages/server/routes/fields.js:315]: adding file filter here is clever. great idea

File: packages/saltcorn-data/models/table.ts [hunk 2]
```diff
@@ -405,7 +405,7 @@ class Table implements AbstractTable {
       for (const trigger of triggers) {
         for (const row of rows) {
           // run triggers on delete
-          await trigger.run!(row);
+          trigger.run!(row);
```
Comment (glutamate) [inline packages/saltcorn-data/models/table.ts:408]: need await here

File: packages/saltcorn-data/models/table.ts [hunk 3]
```diff
@@ -910,18 +910,27 @@ class Table implements AbstractTable {
     filePath: string
   ): Promise<ResultMessage> {
     let rows;
+    const state = await require("../db/state").getState();
     try {
-      const s = await getLines(filePath, 500);
-      rows = await csvtojson().fromString(s); // todo agrument type unknown
+      let lines_limit = state.getConfig("csv_lines_limit");
```
Comment (glutamate) [inline packages/saltcorn-data/models/table.ts:915]: The 500 in the previous code was not the limit to the CSV files to be passed. Instead this is the number of lines that are used to determine the types in the created table. The file is read twice, but on the first run we don't need to traverse all of it

File: packages/saltcorn-data/models/config.ts [hunk 4]
```diff
@@ -593,6 +593,28 @@ const configTypes: ConfigTypes = {
...
   },
+  /** @type {object} */
+  default_files_accept_filter: {
+    type: "String",
+    label: "Default Files accept filter",
+    default: null,
+    blurb: "Specifies a default filter for what file types the user can pick from the file input dialog box. Example is `.doc, text/csv,audio/*,video/*,image/*`",
+  },
+  /** @type {object} */
+  csv_lines_limit: {
```
Comment (glutamate) [inline packages/saltcorn-data/models/config.ts:604]: remove this, see comment on table

File: packages/saltcorn-data/models/table.ts (hunk 3 above)
Comment (AleksandrSokolov) [reply packages/saltcorn-data/models/table.ts:915]: Yes. This is clear, Idea to add parameter - that 500 rows can be not relevant in some cases to determine types in created tables. So I tried to give more flexibility to admin for tune this process. I think that needs to change name of "csv_lines_limit"  to better name. But no idea of better name))

Comment (AleksandrSokolov) [reply packages/saltcorn-data/models/table.ts:915]: I got errors when load big csv files. When 500 rows were not enough to determine types.

File: packages/saltcorn-data/models/table.ts (hunk 2 above)
Comment (AleksandrSokolov) [reply packages/saltcorn-data/models/table.ts:408]: Ok, will fixed soon.

File: packages/server/routes/fields.js (hunk 1 above)
Comment (AleksandrSokolov) [reply packages/server/routes/fields.js:315]: Will fixed soon.

File: packages/saltcorn-data/models/table.ts (hunk 3 above)
Comment (AleksandrSokolov) [reply packages/saltcorn-data/models/table.ts]: @glutamate, name changed to "csv_types_detection_rows". Please review

File: packages/saltcorn-data/models/table.ts (hunk 2 above)
Comment (AleksandrSokolov) [reply packages/saltcorn-data/models/table.ts]: fixed

File: packages/saltcorn-data/models/config.ts (hunk 4 above)
Comment (AleksandrSokolov) [reply packages/saltcorn-data/models/config.ts]: @glutamate, Please see my comments in discussion about table

**Comment by glutamate** [comment]
Thank you, looks good!

**Pull request merged**